### ISSUE TITLE
Automatically warn people and delete msg for certain things 

### ISSUE THREAD

**Issue by dominikaaaa** [kami-blue/bot#22]
Obviously add a blacklist, eg if the sender has roles (excluding Updates as anybody can get it), or if they're explaining that it isn't allowed

- [ ] Discord invites (regex below)
- [ ] Hacks / Cheats
- [ ] UFAs and stuff (rule 3)
- [ ] Repetitive messages (ie spam)
- [ ] Slurs 

```
regex: Todo
```

**Comment by humboldt123** [comment]
What constitutes as a "slur"? Ilidumb frequently uses the n-word without the hard R. Also, do we need to substitute 1 for I to prevent bypasses?

**Comment by dominikaaaa** [comment]
I'll write the regex for the discord links. I think hard r would, I can make a regex for it later

**Comment by humboldt123** [comment]
Not just racial slurs tho. What anarchy players use to refer to "newer players" could be considered a homophobic slur. There are people who will spam transphobic slurs. Alot of this can be consider part of the anarchy experience and "not a slur" but I would disagree. What are your thoughts?

**Comment by dominikaaaa** [comment]
Yea, I agree that those definitely are slurs. We already warn / ban for repeated offenses

**Comment by humboldt123** [comment]
But the frequency of use and seeing how they are part of what most consider to be the language, should the bot pick up on it or not?

**Comment by dominikaaaa** [comment]
Yes, warn them that it isn't allowed. I wouldn't want automated bans unless it's several messages in a few seconds that include slurs (spamming slurs), and just delete spam and warn (regular spamming).

I don't care that people use slurs on anarchy, the discord isn't an anarchy server and slurs aren't allowed

**Comment by humboldt123** [comment]
Gotcha.

**Comment by dominikaaaa** [comment]
```regex
(d.{0,3}i.{0,3}s.{0,3}c.{0,3}o.{0,3}r.{0,3}d).{0,4}(gg|com.{0,3}invite)
```
discord invite regex

**Comment by dominikaaaa** [comment]
I noticed this deletes messages containing hacks / cheats. It should only delete for the other detections, not for this one. Also it should give a reason like
```
Hacks / cheats are against Discord TOS (Rules 3 and 9)
```
```
Slurs are against Rule 1
```

**Comment by dominikaaaa** [comment]
Uhhh warning people for the non deletion mode should have a 30 second cooldown imo
